### PR TITLE
fix(button): prevent text jump in disabled button set

### LINKED ISSUE #5555: ButtonSet: Text on primary button jumps when set is disabled
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.
When the button set is changing between disabled and enabled, the text on the primary button has a subtle jump and is further to the left when disabled compared to when it is enabled.

When two buttons are next to each other without using the `.bx--btn-set` class, the whole button actually jumps.

> Is this issue related to a specific component?
Button

> What browser are you working in?
Chrome 80

> What version of the Carbon Design System are you using?
`carbon-components@10.10.1`
`carbon-components-react@7.10.1`

## Steps to reproduce the issue

1. Visit React storybook -> "Set of buttons"
2. Watch the right button when toggling the "disabled" checkbox

## Additional information

carbon-button-set

### PR DESCRIPTION
**Pull request by janhassel**
Closes #5555

#### Changelog

**Changed**

- Replaced `border-left` for disabled buttons in a set to `box-shadow`

#### Testing / Reviewing

1. Launch react storybook -> "Set of buttons"
2. Watch primary button when toggling "disabled" checkbox
3. Text should align

carbon-button-set

### CODE REVIEW

File: packages/components/src/components/button/_button.scss [hunk 1]
```diff
@@ -32,7 +32,7 @@
     + .#{$prefix}--btn--primary.#{$prefix}--btn--disabled,
   .#{$prefix}--btn--tertiary.#{$prefix}--btn--disabled
     + .#{$prefix}--btn--danger.#{$prefix}--btn--disabled {
-    border-left: rem(1px) solid $disabled-03;
+    box-shadow: -#{rem(1px)} 0 0 0 $disabled-03;
```
Comment (tw15egan) [inline packages/components/src/components/button/_button.scss]: Is it possible to just do `rem(-1px)`?

Comment (janhassel) [reply packages/components/src/components/button/_button.scss]: Yes, totally. I don't know why it didn't come to mind 😄

**Review by tw15egan — APPROVED** [review]
Thanks for the fix! 👍 ✅

**Review by joshblack — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
looks good ! 🎉

**Pull request merged**